designs/credentialManagement/design.md

```diff
@@ -0,0 +1,125 @@
+# Credential Management
+
+## Abstract
+
+This document describes the process of managing and switching of AWS credentials within the toolkit.
+
+## Motivation
+
+AWS provides many different ways to retrieve the credentials required to make calls to AWS including but not limited to:
+credential/config files, environment variables, system properties, EC2 metadata APIs. You can also assume roles using 
+a source set of credentials or use MFA. Due to this large matrix of possibilities the toolkit strives to provide an 
+intuitive abstraction layer on top of some of these sources so that the implementation details are hidden and a 
+consistent user experience can be provided, while also enabling future expansion of new sources, including possibly by 
+third party plugins.
+
+## Specification
+
+This toolkit will introduce the following concepts:
+
+1. `ToolkitCredentialsProvider` - An abstract class that implements [AwsCredentialProvider] in the SDK. This class does not
```

Comment:
The class name should be `AwsCredentialsProvider` in the plural format.

Reply:
There is only 1 provider though

Reply:
I mean the `[AwsCredentialProvider] in the SDK`. The class name should be `[AwsCredentialsProvider]`

Reply:
Ah